core/dronecore.cpp

```diff
@@ -38,6 +39,26 @@ DroneCore::ConnectionResult DroneCore::add_udp_connection(int local_port_number)
     return DroneCore::ConnectionResult::SUCCESS;
 }
 
+DroneCore::ConnectionResult DroneCore::add_tcp_connection()
+{
+    return add_tcp_connection("127.0.0.1", 0);
```

Comment:
Do we ever need not localhost? What is the idea of exposing this as an argument?

Reply:
@julianoes : Are we not going to deprecate **discover device** method and replace it with connecting to a specific IP/PORT ? If so, then we need to make similar changes to existing UDP solution too.. Is that right ?

Reply:
No we are not going to deprecate "discover device" as far as I know. Why would we do that? The thing is that you need to add a connection and then wait until you actually receive mavlink.